lib/routes/routeBackbeat.js

```diff
@@ -578,62 +616,99 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
-                    // In that case "removing old data location key" is handled by the lifecycle
-                    // transition processor.
-                    omVal.location && Array.isArray(omVal.location) &&
-                    locationKeysHaveChanged(objMd.location, omVal.location)) {
-                    log.info('removing old data locations', {
-                        method: 'putMetadata',
-                        bucketName,
-                        objectKey,
-                    });
-                    async.eachLimit(objMd.location, 5,
-                    (loc, next) => dataWrapper.data.delete(loc, log, err => {
+                return getCanonicalIdsByAccountId(request.query.accountId, log, (err, res) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    if (res) {
+                        omVal['owner-display-name'] = res.name;
+                        omVal['owner-id'] = res.canonicalId;
+                    }
+                    return next();
+                });
+            },
+            next => {
+                log.trace('putting object version', {
+                    objectKey: request.objectKey, omVal, options });
+                return metadata.putObjectMD(bucketName, objectKey, omVal, options, log,
+                    (err, md) => {
                         if (err) {
-                            log.warn('error removing old data location key', {
+                            log.error('error putting object metadata', {
+                                error: err,
+                                method: 'putMetadata',
+                            });
+                            return next(err);
+                        }
+                        pushReplicationMetric(objMd, omVal, bucketName, objectKey, log);
+                        if (objMd &&
+                            headers['x-scal-replication-content'] !== 'METADATA' &&
+                            versionId &&
+                            // The new data location is set to null when archiving to a Cold site.
+                            // In that case "removing old data location key" is handled by the lifecycle
+                            // transition processor.
+                            omVal.location && Array.isArray(omVal.location) &&
+                            locationKeysHaveChanged(objMd.location, omVal.location)) {
+                            log.info('removing old data locations', {
+                                method: 'putMetadata',
                                 bucketName,
                                 objectKey,
-                                locationKey: loc,
-                                error: err.message,
+                            });
+                            async.eachLimit(objMd.location, 5,
+                            (loc, nextEach) => dataWrapper.data.delete(loc, log, err => {
```

Comment:
Not related to your changes but IIRFC there is the `data.batchDelete` method from the data wrapper, that might help simplifying your function (not blocking, just to share)